### DISCUSSION TITLE
Aint Login Insecure?

### DISCUSSION THREAD

**Discussion by whinee** [fastapi-users/fastapi-users#1403] [Q&A] [answered]
Title. aint login insecure if ur sending the password in plain text to the web server like this:

> Payload (application/x-www-form-urlencoded)
>username=king.arthur@camelot.bt&password=guinevere

**Comment by whinee** [comment]
Im gonna post my custom code which hashes the password client side then sends it to the server as a hash.

**Reply by terabyte128** [reply]
Hashing the password on the client doesn't do anything to improve security. If your server accepts a pre-hashed version of the password, then the hashed version essentially _becomes_ the password. So if someone were able to intercept your network request, they'd be able to steal credentials regardless. You'd need to do something a lot more complex to achieve better security (like generating some kind of different, secure random on the server and sending it with each request so the client can use that to generate unique encrypted passwords). 

See https://stackoverflow.com/questions/3715920/is-it-worth-hashing-passwords-on-the-client-side for more info on this.

**Reply by whinee** [reply]
i just saw this reply. and wow, that actually makes sense. thank u so much!

**Comment by frankie567** [comment, marked as answer]
No, it's not. Actually, most forms are sent like this in 99% of websites out there. The security is provided at a lower level thanks to HTTPS / SSL. That's why it's crucial to use HTTPS in production. Hence, it's unnecessary to encrypt the data on the client side.
